### PR TITLE
Mesh-497 Trusted KYC Service Providers

### PR DESCRIPTION
**Pull request by vkandy**
Polymesh relies on a set of trusted KYC SPs to provide KYC/B claims for all network participants. 

This PR introduces a new module `group` to manage a collection of identities. An instance of `group` is used to manage a collection of KYC service providers. Members are added and removed through governance process.

### CODE REVIEW

File: primitives/src/identity_id.rs [hunk 1]
```diff
@@ -3,6 +3,8 @@ use core::fmt::{Display, Formatter};
 use core::str;
 use rstd::prelude::*;
 use runtime_primitives::traits::Printable;
+#[cfg(feature = "std")]
+use serde::{Deserialize, Serialize};
```
Comment (maxsam4) [inline primitives/src/identity_id.rs]: For consistency, let's use `sr_primitives::{Deserialize, Serialize};`

File: runtime/src/group.rs [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+    }
+}
+
+decl_event!(
+	pub enum Event<T, I=DefaultInstance> where
+		<T as system::Trait>::AccountId,
+		<T as Trait<I>>::Event,
+	{
+		/// The given member was added; see the transaction for who.
+		MemberAdded,
```
Comment (maxsam4) [inline runtime/src/group.rs]: Events should include the identity of member affected

File: runtime/src/group.rs [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+		/// The given member was removed; see the transaction for who.
+		MemberRemoved,
+		/// Two members were swapped; see the transaction for who.
+		MembersSwapped,
+		/// The membership was reset; see the transaction for who the new set is.
+		MembersReset,
+		/// Phantom member, never used.
+		Dummy(rstd::marker::PhantomData<(AccountId, Event)>),
```
Comment (maxsam4) [inline runtime/src/group.rs]: Why is this required? Can we remove it?

File: runtime/src/group.rs [hunk 4]
```diff
@@ -0,0 +1,392 @@
...
+//! - Swam members
+//! - Reset group members
+//!
+//! ### Dispatchable Functions
+//!
+//! - `add_member` - Adds a new identity to the group.
+//! - `remove_member` - Remove identity from the group if it exists.
+//! - `swap_member` - Replace one identity with the other.
+//! - `reset_members` - Re-initialize group members.
+//!
```
Comment (maxsam4) [inline runtime/src/group.rs]: Since this is the modified version of the membership module, we should include the relevant license and context at the top.

File: runtime/src/group.rs (hunk 3 above)
Comment (satyamakgec) [inline runtime/src/group.rs]: If we are not using AccountId and Event type in the event then we can remove then from the definition of the Event enum, so we don't require `Dummy()` event anymore.

File: primitives/src/identity_id.rs (hunk 1 above)
Comment (vkandy) [reply primitives/src/identity_id.rs]: Agreed. I've replaced `serde` serializers with `sr-primitive`'s.

File: runtime/src/group.rs (hunk 2 above)
Comment (vkandy) [reply runtime/src/group.rs]: Makes sense. Latest commit should reflect that.

File: runtime/src/group.rs (hunk 3 above)
Comment (vkandy) [reply runtime/src/group.rs]: No this is needed. Because this is an instantiable module, we need 2 type parameters. However, since there are no generic types in the event types, a marker type is used to satisfy rust compiler. It's the same trick used elsewhere too.

File: runtime/src/group.rs (hunk 4 above)
Comment (vkandy) [reply runtime/src/group.rs]: @adamdossa I'll let you make a call on whether or not we need license for this module. This module is definitely inspired by substrate's `membership` module and the dispatchables take the similar function signatures. However the context and intent is different and it isn't just "extending" `membership` module by adding functions to it like we did with `staking`.

File: runtime/src/group.rs (hunk 3 above)
Comment (vkandy) [reply runtime/src/group.rs]: This is done on purpose. See comments above.

File: runtime/src/group.rs (hunk 4 above)
Comment (adamdossa) [reply runtime/src/group.rs]: I will check w/ Jesse what the appropriate approach is.
